### PR TITLE
Add boulder

### PR DESCRIPTION
**Pull request by PulexIrritans**
Now, new boulder routes can be added to the db by the user. I also managed to implement a redirect to the "Add" page on successful boulder creation. 
Some styling and a11y features for this are implemented as well as error handling.
I also did some code refactoring and created a new error component which I used here.

### CODE REVIEW

File: client/src/components/AddBoulder.jsx [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+const URL = process.env.REACT_APP_URL;
+
+const AddBoulder = () => {
+  const [error, setError] = useState(false);
+  const [newBoulder, setNewBoulder] = useState({});
+  const [addedBoulder, setAddedBoulder] = useState({});
+
+  const navigate = useNavigate();
+
+  const toTitleCase = text => {
```
Comment (m4xnull) [inline client/src/components/AddBoulder.jsx:16]: props for taking the extra mile ;)

File: client/src/components/AddBoulder.jsx [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+  };
+
+  useEffect(() => {
+    addedBoulder._id && navigate(`/add/${addedBoulder._id}`);
+  }, [addedBoulder]);
+
+  return (
+    <AddBoulderFormWrapper>
+      <FormHeader id="form-heading">
+        Couldn't find the boulder you want to climb?
```
Comment (m4xnull) [inline client/src/components/AddBoulder.jsx:61]: Maybe rather "boulder you want to record" or "boulder you climbed"?

File: client/src/components/TextInput.jsx [hunk 3]
```diff
@@ -14,7 +15,7 @@ const TextInput = ({
     <Wrapper>
       <Label htmlFor={title}>{title}</Label>
       <TextFieldInput
-        type="text"
+        type={'text'}
```
Comment (m4xnull) [inline client/src/components/TextInput.jsx:18]: should be {type}, shouldn't it?

**Review by m4xnull — CHANGES_REQUESTED** [review]
Cool new feature and nicely and thoroughly done! Not sure about the text input type, please check.

File: client/src/components/AddBoulder.jsx (hunk 1 above)
Comment (PulexIrritans) [reply client/src/components/AddBoulder.jsx:16]: Thank you! :-)

**Comment by PulexIrritans** [comment]
I did some minor improvements based on your comments. Ready to approve now? :-)

**Review by m4xnull — APPROVED** [review]
great. my bad regarding type parameter..

**Pull request merged**